### DISCUSSION TITLE
Auto reload from Bandolier

### DISCUSSION THREAD

**Discussion by Marutzo** [FakeFishGames/Barotrauma#12694] [Ideas and suggestions]
Since the Bandolier doesn't offer much in terms of armor and you can't use it in / on a diving suit, how about letting the wearer auto reload from its item slots? If you have bullets or magazines in it, and you fire a gun until you have no ammo left, the vest either:

Inserts bullets from itself into the empty gun.

Or in case of magazines: swaps a full magazine for the now empty one that is in the gun.

If you have a stack of magazines in the belt, it would naturally be unable to put the empty magazine back in the belt if there was no room for it, so... eject it into the player inventory, and if no room for it even there, drop the empty mag on the ground?

For game balance I would also suggest adding a delay between a vest reload and being able to fire the gun again.

**Comment by bookslayer10** [comment]
I think that the hidden increase in ammo magazine stack size, 25% buff to submarine weapon firing speed, and 25% buff to ranged handweapon firing speed are all enough to make it worth using in dry-land combat situations or while using submarine weapons.
